lib/tools.js

```diff
@@ -173,12 +173,24 @@ function prettifierMetaWrapper (pretty, dest) {
     lastMsg: null,
     lastObj: null,
     lastLogger: null,
+    get destStreamContext () {
+      if (dest[needsMetadataGsym] === true) {
+        return Object.assign(dest, {
+          lastLevel: this.lastLevel,
+          lastMsg: this.lastMsg,
+          lastObj: this.lastObj,
+          lastTime: this.lastTime,
+          lastLogger: this.lastLogger
+        })
+      }
+      return dest
+    },
     flushSync () {
       if (warned) {
         return
       }
       warned = true
-      dest.write(pretty(Object.assign({
+      dest.write.call(this.destStreamContext, pretty(Object.assign({
```

Comment:
Copying streams metadata might cause _significant issues_. I would just attach the properties to the stream itself, mutating it.

Reply:
Fixed it

Reply:
I think you can do something like:

```js
function setMetadataProps (dest, that) {
      if (dest[needsMetadataGsym] === true) {
        dest.lastLevel = that.lastLevel
        dest.lastMsg = that.lastMsg
        dest.lastObj = that.lastObj
        dest.lastTime = that.lastTime
        dest.lastLogger = that.lastLogger
      }
}
```

and then call in both places.

Reply:
Created setMetadataProps function